source/Octopus.Client/Editors/Async/TenantEditor.cs

```diff
@@ -42,6 +43,12 @@ public async Task<TenantEditor> CreateOrModify(string name)
         
         public async Task<TenantEditor> CreateOrModify(string name, string description, string cloneId = null)
         {
+            var baseRepository = ((TenantRepository) repository).Repository;
```

Comment:
If this component specifically requires a `TenantRepository` instead of any `ITenantRepository`, it should probably depend on that directly. Or this property be exposed in `ITenantRepository`?

Reply:
Agreed this is a bit of a smell.
However, I'm not entirely sure what the best approach is to tackle it... Depending on `TenantRepository` directly cant be done, as all the implementations are internal - only interfaces are exposed. Adding that property to ITenantRepository is also a bit iffy as it exposes stuff that's meant to be internal.
I'm inclined to go "it's not ideal" and move on. I dont think there's much chance this will come back and bite us.